### DISCUSSION TITLE
Make a unified import and export functions in commons and unified file type too (maybe?)

### DISCUSSION THREAD

**Discussion by ronniedroid** [FossifyOrg/General-Discussion#106] [Ideas]
There is too much redundant functions doing the same thing in a not unified way, for example 

the Messages exports messages as json, numbers as txt with comma separated items

The keyboard exports clipboard ad txt with comma separated items  but in an array

The Dialer exports call history as json.

The gallery exports the faviores also as txt but with new line seperated items.

I think we should discuss a unified way, preferably have it in commons to export and import data from the apps and decide on one file type. 

PS: Some apps have an option to export settings and some don't, maybe we ahould unify that too? 

PSS: Last night I had a dream were I build a "Fossify backups" app that could export the settings and data of all installed fossify apps in one go, kinda neat.

**Comment by serrq** [comment]
I need only the phone apps package as unified (phone dialer, messages, contacts book).
But it is very similar to my issue.

**Comment by DeMiro5001** [comment]
May be a unified backup destination.
Or even better, a synchronisation server. The thing that I am struggling to replace is the Google synchronisation. Everything is synced to one account.
The easiest way is to have an sftp server with structured folders for each app. Every newbie can setup a simple sftp server, at home (syncing restricted to home ssid) or remote.
Or using a nextcloud server, that is used by some apps (if I'm not wrong). Syncthing is another option.

**Reply by ronniedroid** [reply]
I was talking more about the back-end than adding a user facing feature.

Like adding some views and helpers to commons that make this unified.

**Comment by serrq** [comment]
I don't want "internet permission". One backup, multiple Fossify apps.

**Comment by MusicNote-Haven** [comment]
First and foremost, I want to express my great appreciation for your work. I am very happy to see an alternative to commercial software; I recently ditched Microsoft ($100+/year) and Google ($35/year) in favor of open-source solutions, including running my own Nextcloud server for storage and Office functionality. I purchased the Fossify Thank You! app to contribute and show my support. I am willing to pay more as the applications mature.

I have installed all available Fossify apps via the Google Play Store.
1. Specific Feedback: Fossify Launcher (org.fossify.home)
I attempted to switch from the MS Launcher to the Fossify Launcher but reverted back after less than an hour due to the following critical points:
- Manual Folder Re-creation: Re-creating all my folders is simply too time-consuming. With over 250 apps on my phone, the effort required to manually reorganize everything is a significant barrier to adoption.
- Lack of Backup Function: The absence of a backup/restore function for the home screen layout is a major deterrent. The risk of losing all that configuration work is too high.
- App Selection for Folders: When creating a new folder (e.g., "Social"), it is cumbersome that I can only select apps one by one. The ability to select multiple apps simultaneously would dramatically speed up the setup process.

2. General Improvements & Integration Wishes
A common theme across my feedback is the desire for better integration with local and self-hosted storage solutions like Nextcloud/TrueNAS.
- File Manager (org.fossify.filemanager): Support for SMB/CIFS-shares is highly desired, as all my files reside on my TrueNAS SCALE server.
- Music Player (org.fossify.musicplayer): Same request: The ability to add/select an SMB share as a music source.
- Notes (org.fossify.notes): An option to sync with Nextcloud Notes would be excellent. Until this is available, I will continue to use the official Nextcloud Notes app.
- Camera (org.fossify.camera): I am curious about the integration and sync mechanism with my Nextcloud Photos or Memories app.
- Keyboard (org.fossify.keyboard): I switched back to MS SwiftKey because I miss the dedicated arrow key row to move the cursor. Also, the predictive text features do not seem to work optimally (or I have not found the settings).

Current Usage:
I am actively using the following apps: Thank You!, Messages, Clock, Calculator, Phone. I will test the others as they mature.

3. Offer of Assistance
- Testing: If you need an active tester, I am happy to dedicate my time to help.
- Translation: I can assist with the Dutch translation (manual, contextually accurate translation) to ensure quality.

Thank you again for all your hard work!

**Reply by naveensingh** [reply]
Glad you like it. Some general rules and information regarding translations are here: 
 - https://github.com/FossifyOrg/General-Discussion#contributing-to-fossify
 - https://github.com/FossifyOrg/General-Discussion#contributing-translations

Some feature requests related to your suggestions:
 - https://github.com/FossifyOrg/Launcher/issues/94
 - https://github.com/FossifyOrg/File-Manager/issues/50
 - https://github.com/FossifyOrg/Keyboard/issues/58

Features that depend on the internet are not yet possible because all Fossify apps work offline (by design). There may be flavors with internet access, but there's no ETA on that.
